### PR TITLE
example url

### PR DESCRIPTION
**Pull request by pca006132**
1. Allows choosing the script via URL hash, e.g. `#Intro`.
2. Removed the annoying newline at the start of examples.

### CODE REVIEW

**Comment by pca006132** [comment]
I think we can also allow users to import raw code from another website, e.g. gist, but not sure what is the best way of embedding that into the url.

File: bindings/wasm/examples/editor.js [hunk 1]
```diff
@@ -299,7 +301,14 @@ require(['vs/editor/editor.main'], async function() {
       addEdit(button);
     }
   }
-  switchTo(currentName);
+  if (window.location.hash.length > 0) {
+    const name = unescape(window.location.hash.substring(1));
+    if (exampleFunctions.get(name) != null || getScript(name) != null)
+      switchTo(name);
```
Comment (elalish) [inline bindings/wasm/examples/editor.js:307]: Is this `if` necessary?

**Review by elalish — APPROVED** [review]
Thanks!

File: bindings/wasm/examples/editor.js (hunk 1 above)
Comment (pca006132) [reply bindings/wasm/examples/editor.js:307]: yeah it is not really needed, will remove

**Comment by pca006132** [comment]
OK nice, will do that.

**Comment by pca006132** [comment]
I thought about this, I don't think this is usually in the threat model. For example, if you think about sites like codepen, they also allow user submitted js. The typical issue is usually things like stealing credentials or cookies, but we don't have that here. We do need to think if we consider user code as confidential.

**Comment by pca006132** [comment]
Added script loading code, now users can load code via `?url=${url}&name=${scriptName}`.

File: bindings/wasm/examples/editor.js [hunk 2]
```diff
@@ -299,7 +302,25 @@ require(['vs/editor/editor.main'], async function() {
       addEdit(button);
     }
   }
-  switchTo(currentName);
+  const params = new URLSearchParams(window.location.search);
+  if (window.location.hash.length > 0) {
+    const name = unescape(window.location.hash.substring(1));
+    switchTo(name);
+  } else if (params.get('url') != null) {
+    console.log(`Fetching ${params.get('url')}`);
```
Comment (elalish) [inline bindings/wasm/examples/editor.js:310]: It seems weird to call a url param `url`, especially when it's not actually a URL. How about `script`?

File: bindings/wasm/examples/editor.js [hunk 3]
```diff
@@ -299,7 +302,25 @@ require(['vs/editor/editor.main'], async function() {
...
+    autoExecute = false;
+    const response = await fetch(params.get('url'));
+    const code = await response.text();
+    const name = uniqueName(params.get('name') ?? 'New Script');
+    setScript(name, code);
+    const nextButton = createDropdownItem(name);
+    newButton.insertAdjacentElement('afterend', nextButton.parentElement);
+    addEdit(nextButton);
+    switchTo(name);
```
Comment (elalish) [inline bindings/wasm/examples/editor.js:319]: Is some of this duplicated code that could be a function?

**Review by elalish — COMMENTED** [review]
Yeah, I guess I'm just used to Google-level security review, which our site certainly doesn't need. Might be cool to add a little share button that created this link with the proper escaping.

File: bindings/wasm/examples/editor.js (hunk 2 above)
Comment (pca006132) [reply bindings/wasm/examples/editor.js:310]: sure

File: bindings/wasm/examples/editor.js (hunk 3 above)
Comment (pca006132) [reply bindings/wasm/examples/editor.js:319]: yes

**Comment by pca006132** [comment]
yeah I wanted to add a UI control for this, but I don't have a good design for this.

**Comment by elalish** [comment]
> yeah I wanted to add a UI control for this, but I don't have a good design for this.

I'll take care of it eventually. Basically just a little circle button with a share icon somewhere.

**Comment by pca006132** [comment]
Nice. Will do the changes later this week as I have a deadline this Thursday night. You can also take over this if you like.

**Comment by elalish** [comment]
No rush, sounds good.

**Review by elalish — APPROVED** [review]
Thanks!

**Pull request merged**